### ISSUE TITLE
Handle missing datastores by displaying user-readable error message

### ISSUE THREAD

**Issue by cormachogan** [vmware/vic#1259]
**VIC Version:**

0.1 / master / #1221 

**Code**

```ts
// A self-contained demonstration of the problem follows...

This is using a physical ESXi 6.0U2 host as a target

When the datastore name is misspelt in vic-machine-linux create, the error returned is not intuitive:

<<---misspelt datastore name--->>>

root@photon-NaTv5i8IA [ /workspace/vic/bin ]# ./vic-machine-linux create --image-datastore isilion-nfs -t "root@10.27.51.8"
INFO[2016-06-29T13:14:05Z] Please enter ESX or vCenter password:
INFO[2016-06-29T13:14:09Z] ### Installing VCH ####
INFO[2016-06-29T13:14:09Z] Generating certificate/key pair - private key in ./docker-appliance-key.pem
INFO[2016-06-29T13:14:10Z] Validating supplied configuration
vic-machine-linux failed:  runtime error: invalid memory address or nil pointer dereference

<<---correctly spelt datastore name--->>>

root@photon-NaTv5i8IA [ /workspace/vic/bin ]# ./vic-machine-linux create --image-datastore isilion-nfs-01 -t "root@10.27.51.8"
INFO[2016-06-29T13:14:25Z] Please enter ESX or vCenter password:
INFO[2016-06-29T13:14:31Z] ### Installing VCH ####
INFO[2016-06-29T13:14:31Z] Generating certificate/key pair - private key in ./docker-appliance-key.pem
INFO[2016-06-29T13:14:31Z] Validating supplied configuration
INFO[2016-06-29T13:14:32Z] Firewall status: DISABLED on /ha-datacenter/host/esxi-hp-08.rainpole.com/esxi-hp-08.rainpole.com
INFO[2016-06-29T13:14:32Z] Firewall configuration OK on hosts:
INFO[2016-06-29T13:14:32Z]   /ha-datacenter/host/esxi-hp-08.rainpole.com/esxi-hp-08.rainpole.com
INFO[2016-06-29T13:14:32Z] License check OK
INFO[2016-06-29T13:14:33Z] Creating Resource Pool docker-appliance
INFO[2016-06-29T13:14:33Z] Creating VirtualSwitch
INFO[2016-06-29T13:14:33Z] Creating Portgroup
INFO[2016-06-29T13:14:33Z] Creating appliance on target
INFO[2016-06-29T13:14:33Z] Network role client is sharing NIC with external
INFO[2016-06-29T13:14:33Z] Network role management is sharing NIC with external
INFO[2016-06-29T13:14:33Z] Uploading images for container
INFO[2016-06-29T13:14:33Z]      bootstrap.iso
INFO[2016-06-29T13:14:33Z]      appliance.iso
INFO[2016-06-29T13:14:38Z] Waiting for IP information
INFO[2016-06-29T13:14:49Z] Waiting for major appliance components to launch
INFO[2016-06-29T13:14:50Z] Initialization of appliance successful
INFO[2016-06-29T13:14:50Z]
INFO[2016-06-29T13:14:50Z] SSH to appliance (default=root:password)
INFO[2016-06-29T13:14:50Z] ssh root@10.27.33.181
INFO[2016-06-29T13:14:50Z]
INFO[2016-06-29T13:14:50Z] Log server:
INFO[2016-06-29T13:14:50Z] https://10.27.33.181:2378
INFO[2016-06-29T13:14:50Z]
INFO[2016-06-29T13:14:50Z] DOCKER_HOST=10.27.33.181:2376
INFO[2016-06-29T13:14:50Z] DOCKER_OPTS="-H 10.27.33.181:2376 --tls"
INFO[2016-06-29T13:14:50Z]
INFO[2016-06-29T13:14:50Z] Connect to docker:
INFO[2016-06-29T13:14:50Z] docker -H 10.27.33.181:2376 --tls info
INFO[2016-06-29T13:14:50Z] Installer completed successfully
root@photon-NaTv5i8IA [ /workspace/vic/bin ]#

```

**Expected behavior:**

If the datastore name is not spelt, the command should return "no such datastore" or something similar.

**Actual behavior:**

vic-machine-linux failed:  runtime error: invalid memory address or nil pointer dereference

**Comment by andrewtchin** [comment]
Hi @cormachogan - could you verify what version you attempted this against? 
As of commit d2134331a4c4a1b99836a292f90ac9a41c59f412 (June 28) I was unable to repro this against ESX or vCenter:

ESX:
```
INFO[2016-06-29T12:35:11-07:00] ### Installing VCH ####                      
INFO[2016-06-29T12:35:11-07:00] Generating certificate/key pair - private key in ./chin-vic-000-key.pem 
INFO[2016-06-29T12:35:11-07:00] Validating supplied configuration            
INFO[2016-06-29T12:35:11-07:00] Firewall status: DISABLED on /ha-datacenter/host/localhost.localdomain/localhost.localdomain 
INFO[2016-06-29T12:35:11-07:00] Firewall configuration OK on hosts:          
INFO[2016-06-29T12:35:11-07:00]   /ha-datacenter/host/localhost.localdomain/localhost.localdomain 
WARN[2016-06-29T12:35:11-07:00] Evaluation license detected. VIC may not function if evaluation expires or insufficient license is later assigned. 
INFO[2016-06-29T12:35:11-07:00] License check OK                             
INFO[2016-06-29T12:35:11-07:00] DRS check SKIPPED - target is standalone host 
ERRO[2016-06-29T12:35:11-07:00] datastore 'datastore1asdf' not found         
ERRO[2016-06-29T12:35:11-07:00] Creation cannot continue: configuration validation failed 
vic-machine-linux failed:  validation of configuration failed
```

vCenter:
```
INFO[2016-06-29T12:34:15-07:00] ### Installing VCH ####                      
INFO[2016-06-29T12:34:15-07:00] Generating certificate/key pair - private key in ./chin-vic-005-key.pem 
INFO[2016-06-29T12:34:16-07:00] Validating supplied configuration            
INFO[2016-06-29T12:34:17-07:00] Firewall status: ENABLED on /dc1/host/cluster1/10.17.109.109 
INFO[2016-06-29T12:34:17-07:00] Firewall status: ENABLED on /dc1/host/cluster1/10.17.109.143 
INFO[2016-06-29T12:34:18-07:00] Firewall status: ENABLED on /dc1/host/cluster1/10.17.109.183 
INFO[2016-06-29T12:34:18-07:00] Firewall configuration OK on hosts:          
INFO[2016-06-29T12:34:18-07:00]   /dc1/host/cluster1/10.17.109.109           
INFO[2016-06-29T12:34:18-07:00]   /dc1/host/cluster1/10.17.109.143           
INFO[2016-06-29T12:34:18-07:00]   /dc1/host/cluster1/10.17.109.183           
INFO[2016-06-29T12:34:18-07:00] License check OK on hosts:                   
INFO[2016-06-29T12:34:18-07:00]   /dc1/host/cluster1/10.17.109.109           
INFO[2016-06-29T12:34:18-07:00]   /dc1/host/cluster1/10.17.109.143           
INFO[2016-06-29T12:34:18-07:00]   /dc1/host/cluster1/10.17.109.183           
INFO[2016-06-29T12:34:18-07:00] DRS check OK on:                             
INFO[2016-06-29T12:34:18-07:00]   /dc1/host/cluster1/Resources               
ERRO[2016-06-29T12:34:18-07:00] datastore 'vsanDatastoreasdf' not found      
ERRO[2016-06-29T12:34:18-07:00] Creation cannot continue: configuration validation failed 
vic-machine-linux failed:  validation of configuration failed
```

**Comment by cormachogan** [comment]
Hey @andrewtchin - the error is still happening for me and I am using build #2347 from bintray - https://bintray.com/vmware/vic-repo/build/view - this was the latest build from yesterday afternoon that pass testing. My datastore is isilion-nfs-01, but if I just put in isilion-nfs, the error occurs:

root@photon-NaTv5i8IA [ /workspace/vic ]#  ./vic-machine-linux create  --bridge-network Bridge-DPG --image-datastore isilion-nfs  -t "administrator@vsphere.local@10.27.51.103" --compute-resource Mgmt
INFO[2016-06-30T07:20:38Z] Please enter ESX or vCenter password:
INFO[2016-06-30T07:20:41Z] ### Installing VCH ####
INFO[2016-06-30T07:20:41Z] Generating certificate/key pair - private key in ./docker-appliance-key.pem
INFO[2016-06-30T07:20:42Z] Validating supplied configuration
vic-machine-linux failed:  runtime error: invalid memory address or nil pointer dereference

**Comment by andrewtchin** [comment]
I tested both these commands on build 2496 which includes the fix from #1300 and found this to be resolved.

```
./vic-machine-linux create --image-datastore isilion-nfs -t "root@10.27.51.8" --name chin-vic-003
INFO[2016-07-01T16:30:56Z] Please enter ESX or vCenter password:        
INFO[2016-07-01T16:31:01Z] ### Installing VCH ####                      
INFO[2016-07-01T16:31:01Z] Generating certificate/key pair - private key in ./chin-vic-003-key.pem 
INFO[2016-07-01T16:31:01Z] Validating supplied configuration            
INFO[2016-07-01T16:31:01Z] isilion-nfs                                  
ERRO[2016-07-01T16:31:01Z] Datastore for images is nil                  
ERRO[2016-07-01T16:31:01Z] Firewall check SKIPPED                       
ERRO[2016-07-01T16:31:01Z]   datastore not set                          
ERRO[2016-07-01T16:31:01Z] License check SKIPPED                        
ERRO[2016-07-01T16:31:01Z]   datastore not set                          
ERRO[2016-07-01T16:31:01Z] DRS check SKIPPED                            
ERRO[2016-07-01T16:31:01Z]   datastore not set                          
ERRO[2016-07-01T16:31:01Z] Compatibility check SKIPPED                  
ERRO[2016-07-01T16:31:01Z]   datastore not set                          
ERRO[2016-07-01T16:31:01Z] --------------------                         
ERRO[2016-07-01T16:31:01Z] datastore 'isilion-nfs' not found            
ERRO[2016-07-01T16:31:01Z] Firewall check SKIPPED                       
ERRO[2016-07-01T16:31:01Z] License check SKIPPED                        
ERRO[2016-07-01T16:31:01Z] DRS check SKIPPED                            
ERRO[2016-07-01T16:31:01Z] Compatibility check SKIPPED                  
ERRO[2016-07-01T16:31:01Z] Creation cannot continue: configuration validation failed 
ERRO[2016-07-01T16:31:01Z] vic-machine-linux failed: validation of configuration failed 
```

```
./vic-machine-linux create --image-datastore isilion-nfs -t "administrator@vsphere.local@10.27.51.103" --name chin-vic-003
INFO[2016-07-01T16:33:00Z] Please enter ESX or vCenter password:        
INFO[2016-07-01T16:33:03Z] ### Installing VCH ####                      
INFO[2016-07-01T16:33:03Z] Generating certificate/key pair - private key in ./chin-vic-003-key.pem 
INFO[2016-07-01T16:33:04Z] Validating supplied configuration            
INFO[2016-07-01T16:33:04Z] Suggesting valid values for --compute-resource based on * 
INFO[2016-07-01T16:33:04Z]   Cloud                                      
INFO[2016-07-01T16:33:04Z]   Mgmt                                       
INFO[2016-07-01T16:33:04Z] isilion-nfs                                  
ERRO[2016-07-01T16:33:04Z] Datastore for images is nil                  
ERRO[2016-07-01T16:33:04Z] Firewall check SKIPPED                       
ERRO[2016-07-01T16:33:04Z]   datastore not set                          
ERRO[2016-07-01T16:33:04Z]   cluster not set                            
ERRO[2016-07-01T16:33:04Z] License check SKIPPED                        
ERRO[2016-07-01T16:33:04Z]   datastore not set                          
ERRO[2016-07-01T16:33:04Z]   cluster not set                            
ERRO[2016-07-01T16:33:04Z] DRS check SKIPPED                            
ERRO[2016-07-01T16:33:04Z]   datastore not set                          
ERRO[2016-07-01T16:33:04Z]   cluster not set                            
ERRO[2016-07-01T16:33:04Z] Compatibility check SKIPPED                  
ERRO[2016-07-01T16:33:04Z]   datastore not set                          
ERRO[2016-07-01T16:33:04Z]   cluster not set                            
ERRO[2016-07-01T16:33:04Z] --------------------                         
ERRO[2016-07-01T16:33:04Z] No unambiguous default compute resource available: --compute-resource must be specified 
ERRO[2016-07-01T16:33:04Z] datastore 'isilion-nfs' not found            
ERRO[2016-07-01T16:33:04Z] An existing distributed port group must be specified for bridge network on vCenter: network 'chin-vic-003' not found 
ERRO[2016-07-01T16:33:04Z] Firewall check SKIPPED                       
ERRO[2016-07-01T16:33:04Z] License check SKIPPED                        
ERRO[2016-07-01T16:33:04Z] DRS check SKIPPED                            
ERRO[2016-07-01T16:33:04Z] Compatibility check SKIPPED                  
ERRO[2016-07-01T16:33:04Z] Creation cannot continue: configuration validation failed 
ERRO[2016-07-01T16:33:04Z] vic-machine-linux failed: validation of configuration failed
```
